### PR TITLE
eigenlayer: init at 0.5.0

### PR DESCRIPTION
**Pull request by selfuryon**
## Description of changes
[EigenLayer](https://www.eigenlayer.xyz/) has a [cli utility](https://github.com/NethermindEth/eigenlayer/tree/develop) to manage operators.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

Merge after https://github.com/NixOS/nixpkgs/pull/265354 because I added myself to maintainer list there

### CODE REVIEW

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (MikaelFangel) [inline pkgs/by-name/ei/eigenlayer-cli/package.nix:6]: 1.21 is the current default of buildGoModule and is the preferred way (:

```suggestion
  buildGoModule,
  fetchFromGitHub,
}:
buildGoModule rec {
```

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (MikaelFangel) [inline pkgs/by-name/ei/eigenlayer-cli/package.nix:35]:
```suggestion
    description = "Utility that manages core operator functionalities like local keys, operator registration and updates";
```

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (MikaelFangel) [inline pkgs/by-name/ei/eigenlayer-cli/package.nix:25]:
```suggestion
```

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (MikaelFangel) [inline pkgs/by-name/ei/eigenlayer-cli/package.nix:20]:
```suggestion
  subPackages = [ "cmd/eigenlayer" ];
```

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (MikaelFangel) [inline pkgs/by-name/ei/eigenlayer-cli/package.nix:38]:
```suggestion
    maintainers = with maintainers; [ selfuryon ];
```

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (selfuryon) [reply pkgs/by-name/ei/eigenlayer-cli/package.nix]: Fixed!

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (selfuryon) [reply pkgs/by-name/ei/eigenlayer-cli/package.nix]: Fixed

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (selfuryon) [reply pkgs/by-name/ei/eigenlayer-cli/package.nix]: removed

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (selfuryon) [reply pkgs/by-name/ei/eigenlayer-cli/package.nix]: Added extra spaces (but interesting, cuz my formatter removes them :)

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (selfuryon) [reply pkgs/by-name/ei/eigenlayer-cli/package.nix]: fixed

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (MikaelFangel) [reply pkgs/by-name/ei/eigenlayer-cli/package.nix]: Yeah, I've seen both approaches and my formatters add them, so maybe not that important.

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (WolfangAukang) [inline pkgs/by-name/ei/eigenlayer-cli/package.nix:5]:
```suggestion
{ lib
, buildGoModule
, fetchFromGitHub
}:

```

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (WolfangAukang) [inline pkgs/by-name/ei/eigenlayer-cli/package.nix:22]: Please specify in a comment above this line why the tests are disabled.

**Review by WolfangAukang — CHANGES_REQUESTED** [review]

**Comment by WolfangAukang** [comment]
Also, it is missing its entry on `pkgs/top-level/all-packages.nix`. This is why the checks are not listing the package outputs.

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (selfuryon) [reply pkgs/by-name/ei/eigenlayer-cli/package.nix]: Is that standard? Which formatter do that? Fixed

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (selfuryon) [reply pkgs/by-name/ei/eigenlayer-cli/package.nix:22]: Added

**Comment by selfuryon** [comment]
> Also, it is missing its entry on `pkgs/top-level/all-packages.nix`. This is why the checks are not listing the package outputs.

Here in [README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md?plain=1#L6) has been said that it isn't necessary to add packages from `pkgs/by-name` folder to `all-packages.nix`. It is necessary?

**Comment by MikaelFangel** [comment]
It's the preferred way of adding new packages so no it's not strictly necessary, but it should be done if possible (:

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (WolfangAukang) [inline pkgs/by-name/ei/eigenlayer-cli/package.nix:30]: To make it easier, you can use the `${src.rev}` part to use the complete tag.

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (WolfangAukang) [inline pkgs/by-name/ei/eigenlayer-cli/package.nix:4]:
```suggestion
}:

```

**Review by WolfangAukang — COMMENTED** [review]
Small requests, but LGTM

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (WolfangAukang) [reply pkgs/by-name/ei/eigenlayer-cli/package.nix]: I use `nixpkgs-fmt`, but the main idea is the extra space after the `}:` and the `buildGoModule`

Comment (selfuryon) [reply pkgs/by-name/ei/eigenlayer-cli/package.nix]: Oh, got that, I will try to use `nixpkgs-fmt` for next PRs.

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (selfuryon) [reply pkgs/by-name/ei/eigenlayer-cli/package.nix]: Good point, done

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (selfuryon) [reply pkgs/by-name/ei/eigenlayer-cli/package.nix:4]: Now I got that :) Thanks, done

**Comment by MikaelFangel** [comment]
I get the following error when I try to build it:
```
error: builder for '/nix/store/8zvg4b5rpkpkgp06i5fhgdfps7fv40q1-eigenlayer-cli-0.4.3-go-modules.drv' failed with exit code 1;
       last 10 log lines:
       >        github.com/NethermindEth/eigenlayer/internal/docker/mocks: no required module provides package github.com/NethermindEth/eigenlayer/internal/docker/mocks; to add it:
       >   go get github.com/NethermindEth/eigenlayer/internal/docker/mocks
       > go: github.com/NethermindEth/eigenlayer/pkg/daemon tested by
       >         github.com/NethermindEth/eigenlayer/pkg/daemon.test imports
       >    github.com/NethermindEth/eigenlayer/pkg/daemon/mocks: no required module provides package github.com/NethermindEth/eigenlayer/pkg/daemon/mocks; to add it:
       >     go get github.com/NethermindEth/eigenlayer/pkg/daemon/mocks
       > go: github.com/NethermindEth/eigenlayer/pkg/monitoring tested by
       >  github.com/NethermindEth/eigenlayer/pkg/monitoring.test imports
       >        github.com/NethermindEth/eigenlayer/pkg/monitoring/mocks: no required module provides package github.com/NethermindEth/eigenlayer/pkg/monitoring/mocks; to add it:
       >     go get github.com/NethermindEth/eigenlayer/pkg/monitoring/mocks
       For full logs, run 'nix log /nix/store/8zvg4b5rpkpkgp06i5fhgdfps7fv40q1-eigenlayer-cli-0.4.3-go-modules.drv'.
error: 1 dependencies of derivation '/nix/store/f28lcjzxksmy7xgg6isf9cpn6cs01rfc-eigenlayer-cli-0.4.3.drv' failed to build
```

**Comment by WolfangAukang** [comment]
```
error: undefined variable 'selfuryon'

       at /home/alea/.cache/nixpkgs-review/pr-267696-3/nixpkgs/pkgs/by-name/ei/eigenlayer-cli/package.nix:45:39:

           44|     license = licenses.bsl11;
           45|     maintainers = with maintainers; [ selfuryon ];
             |                                       ^
           46|   };
https://github.com/NixOS/nixpkgs/pull/267696 failed to build: Failed to list packages: nix-env failed with exit code 1
```

You need to add a commit adding yourself to the maintainer list.

**Comment by selfuryon** [comment]
> ```
> error: undefined variable 'selfuryon'
> 
>        at /home/alea/.cache/nixpkgs-review/pr-267696-3/nixpkgs/pkgs/by-name/ei/eigenlayer-cli/package.nix:45:39:
> 
>            44|     license = licenses.bsl11;
>            45|     maintainers = with maintainers; [ selfuryon ];
>              |                                       ^
>            46|   };
> https://github.com/NixOS/nixpkgs/pull/267696 failed to build: Failed to list packages: nix-env failed with exit code 1
> ```
> 
> You need to push a separate commit adding yourself to the maintainer list.

Yeah, I added myself in an another PR (I hoped it will be merged before this one):
> Merge after https://github.com/NixOS/nixpkgs/pull/265354 because I added myself to maintainer list there

**Comment by WolfangAukang** [comment]
Ah got it. In this case I am going to check the other one as it makes this PR dependent of it. Can you set this one as a draft for now until the other one is merged?

**Comment by selfuryon** [comment]
> Ah got it. In this case I am going to check the other one as it makes this PR dependent of it. Can you set this one as a draft for now until the other one is merged?

My previous PR was merged, so now it's ready :)

**Review by WolfangAukang — APPROVED** [review]

**Review by MikaelFangel — APPROVED** [review]

**Review by pbsds — COMMENTED** [review]
Result of `nixpkgs-review pr 267696` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>eigenlayer-cli</li>
  </ul>
</details>

Seems to run fine, just a nit

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (selfuryon) [reply pkgs/by-name/ei/eigenlayer-cli/package.nix]: Good point, I revert it to `eigenlayer`. Indeed, in docs they call it `eigenlayer cli`, but the program is called `eigenlayer` anyway.

File: pkgs/by-name/ei/eigenlayer-cli/package.nix
Comment (selfuryon) [reply pkgs/by-name/ei/eigenlayer-cli/package.nix]: Fixed

**Comment by selfuryon** [comment]
Eigenlayer cli utility has changed the repo to a new one: https://github.com/Layr-Labs/eigenlayer-cli, I refactor the PR to follow the latest changes

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 267696` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>eigenlayer</li>
  </ul>
</details>

- [x] package path fits guidelines
- [x] package name fits guidelines
- [x] package version fits guidelines
- [x] package builds on x86_64-linux
- [x] executables tested on x86_64-linux
- [x] `meta.description` is set and fits guidelines
- [x] `meta.license` fits upstream license
- [x] `meta.platforms` is set
- [x] `meta.maintainers` is set
- [x] build time only dependencies are declared in `nativeBuildInputs`
- [x] source is fetched using the appropriate function
- [x] the list of `phases` is not overridden
- [x] when a phase (like `installPhase`) is overridden it starts with `runHook preInstall` and ends with `runHook postInstall`.
- [x] patches have a comment describing either the upstream URL or a reason why the patch wasn't upstreamed
- [x] patches that are remotely available are fetched rather than vendored

**Pull request merged**
